src/Query/TaskGroupingTree.ts

```diff
@@ -37,34 +37,34 @@ export class IntermediateTaskGroupsStorage extends Map<string[], Task[]> {}
 class TaskGroupingTreeNode extends GroupingTreeNode<Task> {}
 
 /**
- * IntermediateTaskGroups does the initial grouping together of tasks,
+ * TaskGroupingTree does the initial grouping together of tasks,
  * in alphabetical order by group names.
  *
- * It is essentially a thin wrapper around Map - see IntermediateTaskGroupsStorage.
+ * It is essentially a thin wrapper around Map - see {@link TaskGroupingTreeStorage}.
```

Comment:
what's the point of having a wrapper that wraps the generic when the latter is used only in one context? In other words, `TaskGroupingTree` should be `GroupingTree` and `TaskGroupingTreeStorage` should be `TaskGroupingTree` no? =)

Reply:
> what's the point of having a wrapper that wraps the generic when the latter is used only in one context? In other words, `TaskGroupingTree` should be `GroupingTree` and `TaskGroupingTreeStorage` should be `TaskGroupingTree` no? =)

Because I could not figure out an algorithm that did the whole thing in one step!!

I think today I worked out a way to simplify things further, but felt that this refactoring had gone far enough (too far) for one step.

Reply:
Oh, I misunderstood your question.

The tree code came from #893 - so any useful discussion about it will be there.